src/language-html/utils.js

```diff
@@ -416,6 +399,16 @@ function inferScriptParser(node) {
     }
   }
 
+  if (options && isCustomBlock(node, options)) {
+    const { lang, type } = node.attrMap;
+    if (type) {
+      if (type.endsWith("json")) {
+        return "json";
+      }
+    }
+    return lang;
+  }
```

Comment:
Can we reuse the logic above? so we can support [`type="text/markdown"`](https://deploy-preview-8023--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeeBnGACANgQygHMBeAHRAE98BbXCgPjKgSWe23ewAs5dcISDsOEVe-CNgDuEAE64AJgEIKXNVA7MAVvjYaefAUIqBlBkDJDIGsGQIYMgewZAgAyA0OUAyroEB-1VFQB6TDCZRm6HBY2DCUAA5w5CDwAB4wnjT4sgDWChBS-iB+rOoc4kYiImKGkjLyyu6a+uo6enklxiDm1vbObiAB3kG+zAE+2VB1nPr5goWiIKPScooqHdULVbVco43Nto6u7l79vR4+eISkFACuUMlQ6ZkDQysN45qTJdPlc7nDS7p3EmuWG23bLpYPzMEAAGhAEDCMAAltAMMhQElZOkAApJBAIlD4XBSfCUBEQgBGsnwYGScBgAGUwmSYcRkDBZCc4BDuDA6AB1bgwzC0sBwKmY3kwgBuvMoyHAGEJIHpGDgshgqNJRESyAAZjiFRCtBgYgAhUnkylU2hwAAy9Lgmu1rJAepiVPpRFwcAAiicIPBbbgdSBabIFbIpaKWeCA7J6TBOTCFDBuMgABwABghYRRCs5pLCUozQUVoptEIAjl74CqoViQPgMABaFhwBRNiOyOBlmFtlX4NX4X3+hU0GGM5n2jAut2e702pBav32mD4Imx+OJpAAJghTPwMNwLoAwhAaOqUEEAKwRk4KgAqS6xc-9YbgAEkoM3YFSwFHoQBBN9U0I3X7OAAF8QKAA) and `type="text/x-handlebars-template"` etc?